### ISSUE TITLE
Task: Check Developmental issues in variable_validation script

### ISSUE THREAD

**Issue by kristabh** [kristabh/gaze-following-analysis#16]
Consider reviewing developmental issues by hand.  Currently code is commented out

**Comment by melsod** [comment]
I'll take a look.

**Comment by melsod** [comment]
OK, 

Here's a spreadsheet with the ones that in my opinion should be excluded ("EXCLUDE"), should not be excluded ("NOX") or needs further discussion/attention (anything else).
[gazefollowinghearingdevissues.xlsx](https://github.com/kristabh/gaze-following-analysis/files/3167901/gazefollowinghearingdevissues.xlsx)

Note this is not machine-readable in current form. I just copy-pasted from different spreadsheets, so columns don't necessarily match up. But the first couple (lab, subid) should.

I can make something more useful for putting in the code, but wanted to sort out the issues first. Main question is whether exclusions based on physical-only delays/difficulties should be exclusion criteria.

**Comment by kristabh** [comment]
I don't think there's a reason to exclude based on physical-only delays/difficulties.  The only case I would do so is if there was a diagnosed developmental issue that was underlying the delay/difficulty, and I don't see any cases like that.  Do you want to go ahead and turn this into a machine readable document?

**Comment by melsod** [comment]
OK. Then there are three total exclusions I see. One is for a child with Neonatal abstinence syndrome, one with a chromosomal mutation including cognitive delay, and one where the child "can look up only by moving his head up" (physical only but interferes with looking behaviour).

One of these (elp-georgetown 8855) could be "unmarked" as a session-level exclusion (but not sure it's necessary since they will definitely be excluded).

I've added the meta-data spreadsheet (gfhearingdevissues.csv). All the ones with a Y for "hearing_vision" and/or "cognitive_developmental" are included, only the 3 to exclude are marked in the first column with "EXCLUDE". The others are all listed but marked "NOX" - just to be clear that they were reviewed.

Not sure if you want me attempting the R or not (I'm still very green, but it should be straightforward?).

**Comment by kristabh** [comment]
If you want to give it a try go ahead!  The MB1 code is there already but commented out.  If not or you get stuck, I can pick it up.

**Comment by melsod** [comment]
Pasting info from a sidebar email exchange. I was having trouble finding in the original code where in the code the exclusions are happening.

Krista: "Okay, I looked again and the real work seems to happen in the 02_variable_validation script, starting on line 424
participants_td_to_keep <- read_csv("metadata/participants_cog_hearing_exclusions.csv")
I'm not convinced that it's working as expected, however.  I did make a fix in the exclusions spreadsheet because we've now changed the subid format so that it's lab_subid (fixed some issues with two labs having the same subids).  But I think there's still something else..."
